*** ./smali/android/graphics/Typeface.smali	2018-02-09 22:30:17.369749559 +0530
--- /root/prithvi/patchrom/device/temp/new_smali/framework.jar.out/./smali/android/graphics/Typeface.smali	2018-02-09 22:19:47.044278525 +0530
***************
*** 1058,1063 ****
--- 1058,1069 ----
      .param p0, "family"    # Landroid/graphics/FontListParser$Family;
  
      .prologue
+     invoke-static {p0}, Landroid/graphics/TypefaceHook;->makeFontFamily(Landroid/graphics/FontListParser$Family;)Landroid/graphics/FontFamily;
+ 
+     move-result-object v0
+ 
+     return-object v0
+ 
      new-instance v1, Landroid/graphics/FontFamily;
  
      iget-object v3, p0, Landroid/graphics/FontListParser$Family;->lang:Ljava/lang/String;
