*** ./smali/com/android/server/backup/BackupManagerService$PerformAdbBackupTask.smali	2018-02-09 22:28:21.534214399 +0530
--- /root/prithvi/patchrom/device/temp/new_smali/services.jar.out/./smali/com/android/server/backup/BackupManagerService$PerformAdbBackupTask.smali	2018-02-09 22:17:59.104711414 +0530
***************
*** 635,648 ****
      :try_start_0
      iget-object v4, p0, Lcom/android/server/backup/BackupManagerService$PerformAdbBackupTask;->this$0:Lcom/android/server/backup/BackupManagerService;
  
!     # getter for: Lcom/android/server/backup/BackupManagerService;->mPackageManager:Landroid/content/pm/PackageManager;
!     invoke-static {v4}, Lcom/android/server/backup/BackupManagerService;->access$400(Lcom/android/server/backup/BackupManagerService;)Landroid/content/pm/PackageManager;
  
!     move-result-object v4
  
!     const/16 v5, 0x40
  
!     invoke-virtual {v4, v3, v5}, Landroid/content/pm/PackageManager;->getPackageInfo(Ljava/lang/String;I)Landroid/content/pm/PackageInfo;
  
      move-result-object v2
  
--- 635,649 ----
      :try_start_0
      iget-object v4, p0, Lcom/android/server/backup/BackupManagerService$PerformAdbBackupTask;->this$0:Lcom/android/server/backup/BackupManagerService;
  
!     iget-object v4, v4, Lcom/android/server/backup/BackupManagerService;->mContext:Landroid/content/Context;
! 
!     iget-object v5, p0, Lcom/android/server/backup/BackupManagerService$PerformAdbBackupTask;->mOutputFile:Landroid/os/ParcelFileDescriptor;
  
!     invoke-virtual {v5}, Landroid/os/ParcelFileDescriptor;->getFd()I
  
!     move-result v5
  
!     invoke-static {v4, v3, v5}, Lcom/android/server/backup/BackupManagerServiceInjector;->getPackageInfo(Landroid/content/Context;Ljava/lang/String;I)Landroid/content/pm/PackageInfo;
  
      move-result-object v2
  
***************
*** 887,892 ****
--- 888,907 ----
  
      if-nez v3, :cond_5
  
+     move-object/from16 v0, p0
+ 
+     iget-object v3, v0, Lcom/android/server/backup/BackupManagerService$PerformAdbBackupTask;->mOutputFile:Landroid/os/ParcelFileDescriptor;
+ 
+     invoke-virtual {v3}, Landroid/os/ParcelFileDescriptor;->getFd()I
+ 
+     move-result v3
+ 
+     invoke-static {v3}, Lcom/android/server/backup/BackupManagerServiceInjector;->isForceAllowBackup(I)Z
+ 
+     move-result v3
+ 
+     if-nez v3, :cond_5
+ 
      invoke-interface/range {v21 .. v21}, Ljava/util/Iterator;->remove()V
  
      goto :goto_1
***************
*** 946,951 ****
--- 961,980 ----
      :goto_2
      move-object/from16 v0, p0
  
+     iget-object v3, v0, Lcom/android/server/backup/BackupManagerService$PerformAdbBackupTask;->mOutputFile:Landroid/os/ParcelFileDescriptor;
+ 
+     invoke-virtual {v3}, Landroid/os/ParcelFileDescriptor;->getFd()I
+ 
+     move-result v3
+ 
+     invoke-static {v3}, Lcom/android/server/backup/BackupManagerServiceInjector;->isForceAllowBackup(I)Z
+ 
+     move-result v3
+ 
+     if-nez v3, :cond_9
+ 
+     move-object/from16 v0, p0
+ 
      iget-object v3, v0, Lcom/android/server/backup/BackupManagerService$PerformAdbBackupTask;->this$0:Lcom/android/server/backup/BackupManagerService;
  
      invoke-virtual {v3}, Lcom/android/server/backup/BackupManagerService;->deviceIsEncrypted()Z
***************
*** 1082,1087 ****
--- 1111,1130 ----
      :try_start_6
      move-object/from16 v0, p0
  
+     iget-object v3, v0, Lcom/android/server/backup/BackupManagerService$PerformAdbBackupTask;->mOutputFile:Landroid/os/ParcelFileDescriptor;
+ 
+     move-object/from16 v0, p0
+ 
+     iget-object v4, v0, Lcom/android/server/backup/BackupManagerService$PerformAdbBackupTask;->mOutputFile:Landroid/os/ParcelFileDescriptor;
+ 
+     invoke-virtual {v4}, Landroid/os/ParcelFileDescriptor;->getFd()I
+ 
+     move-result v4
+ 
+     invoke-static {v3, v4}, Lcom/android/server/backup/BackupManagerServiceInjector;->writeMiuiBackupHeader(Landroid/os/ParcelFileDescriptor;I)V
+ 
+     move-object/from16 v0, p0
+ 
      iget-object v3, v0, Lcom/android/server/backup/BackupManagerService$PerformAdbBackupTask;->this$0:Lcom/android/server/backup/BackupManagerService;
  
      move-object/from16 v0, p0
***************
*** 1414,1419 ****
--- 1457,1472 ----
  
      iput-object v3, v0, Lcom/android/server/backup/BackupManagerService$PerformAdbBackupTask;->mBackupEngine:Lcom/android/server/backup/BackupManagerService$FullBackupEngine;
  
+     move-object/from16 v0, p0
+ 
+     iget-object v3, v0, Lcom/android/server/backup/BackupManagerService$PerformAdbBackupTask;->mBackupEngine:Lcom/android/server/backup/BackupManagerService$FullBackupEngine;
+ 
+     move-object/from16 v0, p0
+ 
+     iget-object v4, v0, Lcom/android/server/backup/BackupManagerService$PerformAdbBackupTask;->mOutputFile:Landroid/os/ParcelFileDescriptor;
+ 
+     invoke-static {v3, v4}, Lcom/android/server/backup/BackupManagerServiceInjector;->setOutputFileDescriptor(Lcom/android/server/backup/BackupManagerService$FullBackupEngine;Landroid/os/ParcelFileDescriptor;)V
+ 
      if-eqz v20, :cond_12
  
      const-string v3, "Shared storage"
