*** ./smali/com/android/server/backup/BackupManagerService$PerformFullTransportBackupTask$SinglePackageBackupRunner.smali	2018-02-09 22:28:22.298211333 +0530
--- /root/prithvi/patchrom/device/temp/new_smali/services.jar.out/./smali/com/android/server/backup/BackupManagerService$PerformFullTransportBackupTask$SinglePackageBackupRunner.smali	2018-02-09 22:17:59.728708912 +0530
***************
*** 105,110 ****
--- 105,114 ----
      invoke-direct/range {v0 .. v5}, Lcom/android/server/backup/BackupManagerService$FullBackupEngine;-><init>(Lcom/android/server/backup/BackupManagerService;Ljava/io/OutputStream;Ljava/lang/String;Lcom/android/server/backup/BackupManagerService$FullBackupPreflight;Z)V
  
      .local v0, "engine":Lcom/android/server/backup/BackupManagerService$FullBackupEngine;
+     iget-object v1, p0, Lcom/android/server/backup/BackupManagerService$PerformFullTransportBackupTask$SinglePackageBackupRunner;->mOutput:Landroid/os/ParcelFileDescriptor;
+ 
+     invoke-static {v0, v1}, Lcom/android/server/backup/BackupManagerServiceInjector;->setOutputFileDescriptor(Lcom/android/server/backup/BackupManagerService$FullBackupEngine;Landroid/os/ParcelFileDescriptor;)V
+ 
      iget-object v1, p0, Lcom/android/server/backup/BackupManagerService$PerformFullTransportBackupTask$SinglePackageBackupRunner;->mTarget:Landroid/content/pm/PackageInfo;
  
      invoke-virtual {v0, v1}, Lcom/android/server/backup/BackupManagerService$FullBackupEngine;->backupOnePackage(Landroid/content/pm/PackageInfo;)I
