*** ./smali/com/android/server/am/ActivityManagerService.smali	2018-02-09 22:28:24.002204496 +0530
--- /root/prithvi/patchrom/device/temp/new_smali/services.jar.out/./smali/com/android/server/am/ActivityManagerService.smali	2018-02-09 22:18:01.268702736 +0530
***************
*** 19480,19485 ****
--- 19480,19510 ----
      .end local v31    # "msg":Ljava/lang/String;
      .end local v35    # "providerRunning":Z
      :cond_2
+     invoke-static/range {p5 .. p5}, Lmiui/securityspace/XSpaceUserHandle;->isXSpaceUserId(I)Z
+ 
+     move-result v5
+ 
+     if-eqz v5, :cond_miui_3
+ 
+     move-object/from16 v0, p0
+ 
+     move-object/from16 v1, p2
+ 
+     move/from16 v2, p5
+ 
+     invoke-direct {v0, v1, v2}, Lcom/android/server/am/ActivityManagerService;->getProviderInfoLocked(Ljava/lang/String;I)Landroid/content/pm/ProviderInfo;
+ 
+     move-result-object v5
+ 
+     if-nez v5, :cond_miui_3
+ 
+     const/16 p5, 0x0
+ 
+     const/16 v21, 0x0
+ 
+     goto :goto_0
+ 
+     :cond_miui_3
      const/4 v4, 0x0
  
      const/4 v6, 0x0
***************
*** 19787,19792 ****
--- 19812,19843 ----
      move-wide/from16 v1, v38
  
      invoke-direct {v0, v1, v2, v5}, Lcom/android/server/am/ActivityManagerService;->checkTime(JLjava/lang/String;)V
+     
+     if-nez v6, :cond_miui_c
+     
+     invoke-static/range {p5 .. p5}, Lmiui/securityspace/XSpaceUserHandle;->isXSpaceUserId(I)Z
+ 
+     move-result v5
+ 
+     if-eqz v5, :cond_miui_c
+ 
+     const/16 p5, 0x0
+     
+     move/from16 v1, p5
+     
+     invoke-static {}, Landroid/app/AppGlobals;->getPackageManager()Landroid/content/pm/IPackageManager;
+ 
+     move-result-object v5
+ 
+     const/16 v10, 0xc00
+ 
+     move-object/from16 v0, p2
+     
+     invoke-interface {v5, v0, v10, v1}, Landroid/content/pm/IPackageManager;->resolveContentProvider(Ljava/lang/String;II)Landroid/content/pm/ProviderInfo;
+    
+     move-result-object v6
+     
+     :cond_miui_c
      :try_end_2
      .catch Landroid/os/RemoteException; {:try_start_2 .. :try_end_2} :catch_5
      .catchall {:try_start_2 .. :try_end_2} :catchall_0
***************
*** 27059,27064 ****
--- 27110,27123 ----
  
      aput v5, v8, v4
  
+     invoke-static {v6}, Landroid/os/UserHandle;->getUserId(I)I
+ 
+     move-result v4
+ 
+     invoke-static {v4, v8}, Lcom/android/server/am/ActivityManagerServiceInjector;->computeGids(I[I)[I
+ 
+     move-result-object v8
+ 
      .end local v22    # "mountServiceInternal":Landroid/os/storage/MountServiceInternal;
      .end local v24    # "permGids":[I
      .end local v25    # "pm":Landroid/content/pm/IPackageManager;
