*** ./smali/com/android/internal/telephony/cdma/CdmaServiceStateTracker.smali	2018-02-09 22:28:13.998244640 +0530
--- /root/prithvi/patchrom/device/temp/new_smali/telephony-common.jar.out/./smali/com/android/internal/telephony/cdma/CdmaServiceStateTracker.smali	2018-02-09 22:17:53.512733838 +0530
***************
*** 10168,10173 ****
--- 10168,10177 ----
      move-result-object v3
  
      .local v3, "plmn":Ljava/lang/String;
+     invoke-static {p0}, Lcom/android/internal/telephony/ServiceStateTrackerInjector;->getPlmn(Lcom/android/internal/telephony/ServiceStateTracker;)Ljava/lang/String;
+ 
+     move-result-object v3
+ 
      const/4 v2, 0x0
  
      .local v2, "showPlmn":Z
***************
*** 10242,10248 ****
  
      new-instance v6, Landroid/content/Intent;
  
!     const-string v0, "android.provider.Telephony.SPN_STRINGS_UPDATED"
  
      invoke-direct {v6, v0}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
--- 10246,10254 ----
  
      new-instance v6, Landroid/content/Intent;
  
!     invoke-static {p0}, Lcom/android/internal/telephony/ServiceStateTrackerInjector;->getSpnUpdateActionName(Lcom/android/internal/telephony/ServiceStateTracker;)Ljava/lang/String;
! 
!     move-result-object v0
  
      invoke-direct {v6, v0}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
