*** ./smali/android/app/ActivityThread.smali	2018-02-09 22:30:08.713784296 +0530
--- /root/prithvi/patchrom/device/temp/new_smali/framework.jar.out/./smali/android/app/ActivityThread.smali	2018-02-09 22:19:39.820307499 +0530
***************
*** 6345,6350 ****
--- 6345,6356 ----
      invoke-virtual {v2}, Ldalvik/system/VMRuntime;->clearGrowthLimit()V
  
      :goto_a
+     move-object/from16 v0, p1
+ 
+     iget-object v2, v0, Landroid/app/ActivityThread$AppBindData;->appInfo:Landroid/content/pm/ApplicationInfo;
+ 
+     invoke-static {v2}, Lcom/miui/whetstone/app/WhetstoneAppManager;->trimHeapSizeIfNeeded(Landroid/content/pm/ApplicationInfo;)V
+ 
      invoke-static {}, Landroid/os/StrictMode;->allowThreadDiskWrites()Landroid/os/StrictMode$ThreadPolicy;
  
      move-result-object v26
