*** ./smali/android/app/SystemServiceRegistry.smali	2018-02-09 01:20:55.670547937 +0530
--- /root/prithvi/patchrom/device/temp/new_smali/framework.jar.out/./smali/android/app/SystemServiceRegistry.smali	2018-02-09 01:12:15.040481094 +0530
***************
*** 750,755 ****
--- 750,757 ----
  
      invoke-static {v0, v1, v2}, Landroid/app/SystemServiceRegistry;->registerService(Ljava/lang/String;Ljava/lang/Class;Landroid/app/SystemServiceRegistry$ServiceFetcher;)V
  
+     invoke-static {}, Landroid/app/ContextImplInjector;->registerMiuiServices()V
+ 
      return-void
  .end method
  
